src/Akka.Streams.Kafka.Tests/BugFix240SupervisionStrategy.cs

```diff
@@ -0,0 +1,434 @@
+
+            probe.Request(10);
+            for (var i = 0; i < 9; i++)
+            {
+                var message = probe.ExpectNext(TimeSpan.FromSeconds(10)); 
+                Log.Info(message);
+            }
+            probe.Cancel();
+            
+            callCount.Should().Be(2);
+        }
+        
+        [Fact]
+        public async Task Committable_consumer_with_failed_downstream_stage_result_should_be_gapless()
+        {
+            var topic = CreateTopic(1);
+            var group = CreateGroup(1);
+            var topicPartition = new TopicPartition(topic, 0);
+
+            var consumerSettings = ConsumerSettings<Null, string>.Create(Sys, null, null)
+                .WithBootstrapServers(Fixture.KafkaServer)
+                .WithStopTimeout(TimeSpan.FromSeconds(1))
+                .WithProperty("auto.offset.reset", "earliest")
+                .WithProperty("enable.auto.commit", "false")
+                .WithGroupId(group);
+
+            var counter = 0;
+            
+            await Source.From(Enumerable.Range(1, 11))
+                .Select(elem => new ProducerRecord<Null, string>(topicPartition, elem.ToString()))
+                .RunWith(KafkaProducer.PlainSink(ProducerSettings), Materializer);
+            
+            var probe = KafkaConsumer.CommittableSource(consumerSettings, Subscriptions.AssignmentWithOffset(new TopicPartitionOffset(topicPartition, Offset.Unset)))
+                .Select(t =>
+                {
+                    counter++;
+                    if (counter % 7 == 0)
+                        throw new Exception("BOOM!");
+                    return t;
+                })
+                .SelectAsync(1, async elem =>
+                {
+                    await elem.CommitableOffset.Commit();
+                    return elem.Record.Value;
+                })
+                .ToMaterialized(this.SinkProbe<string>(), Keep.Right)
+                .Run(Materializer);
+
+            var messages = new List<string>();
+            probe.Request(11);
+            for (var i = 0; i < 6; i++)
+            {
+                messages.Add(probe.ExpectNext(TimeSpan.FromSeconds(5))); 
+            }
+
+            // stream fails at index 7
+            var err = probe.ExpectEvent();
+            err.Should().BeOfType<TestSubscriber.OnError>();
+            var exception = ((TestSubscriber.OnError)err).Cause;
+            exception.Message.Should().Be("BOOM!");
+
+            // stream should be dead here
+            probe.ExpectNoMsg(TimeSpan.FromSeconds(5));
+            probe.Cancel();
+            
+            // restart dead stream
+            probe = KafkaConsumer.CommittableSource(consumerSettings, Subscriptions.AssignmentWithOffset(new TopicPartitionOffset(topicPartition, Offset.Unset)))
+                .Select(t =>
+                {
+                    counter++;
+                    if (counter % 7 == 0)
+                        throw new Exception("BOOM!");
```

Comment:
Need to add a comment explaining why this doesn't throw again when we re-create the stream.

Reply:
Fixed